### ISSUE TITLE
CI is failing on coverage runs

### ISSUE THREAD

**Issue by elinorbgr** [Smithay/calloop#183]
I'm not sure what may be the cause, but it looks like our current CI fails in some jobs with the error message

```
fatal runtime error: IO Safety violation: owned file descriptor already closed
```

The interesting part is that it seems that the failing jobs are exactly the jobs measuring test coverage. Is there some weird interaction somewhere?

**Comment by notgull** [comment]
Fixed in #185, it looks like the cause was the nightly compiler rather than the coverage run

**Comment by notgull** [comment]
Yeah, it looks like this line was added to stdlib in the latest nightly:

https://github.com/rust-lang/rust/blob/1d1283ed095ed605297b148e1284e1c53290ec91/library/std/src/os/fd/owned.rs#L181

This broke the "invalid FD" test we had earlier.
